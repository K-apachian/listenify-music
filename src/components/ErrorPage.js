import React from "react";
import { Col, Container, Row, Card } from "react-bootstrap";
import errorimg from "../images/error.jpg";
import "../mystyles/ErrorPage.css";

const ErrorPage = () => {
  return (
    <div className="errorpage">
      <Container fluid className="error-container">
        <Row>
          <h2
            style={{
              color: "red",
              textAlign: "center",
              paddingBottom: "20px",
            }}
          >
            <b>ERROR 404 ! UNAUTHORISED</b>
          </h2>
        </Row>
        <Card className="error-card">
          <Row className="m-3">
            <Col md={6} className="errorcard-image">
              <img src={errorimg} style={{ width: "70%", height: "90%" }} />
            </Col>
            <Col md={6}>
              <Card.Body>
                <h2
                  style={{
                    color: "darkcyan",
                    textAlign: "center",
                    paddingTop: "40px",
                  }}
                >
                  <b>Please Login for Authorised Operations</b>
                </h2>
                <h3
                  style={{
                    color: "crimson",
                    textAlign: "center",
                    paddingTop: "40px",
                  }}
                >
                  <i>
                    You Can't Add, Update and Delete Songs,if you aren't logged
                    in !
                  </i>
                </h3>
              </Card.Body>
            </Col>
          </Row>
        </Card>
      </Container>
    </div>
  );
};

export default ErrorPage;
