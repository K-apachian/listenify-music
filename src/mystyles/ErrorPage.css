.errorpage {
  width: auto;
  height: 100vh;
  background-size: cover;
  background-position: center;
  background-attachment: scroll;
}

.error-container {
  padding-top: 50px;
}
.error-card {
  width: 70%;
  margin: auto;
  border: none;
}

@media only screen and (max-width: 1000px) {
  .errorcard-image {
    display: none;
  }
}
