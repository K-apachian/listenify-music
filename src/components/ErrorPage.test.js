import React from "react";
import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import ErrorPage from "./ErrorPage";

const mockStore = configureMockStore();
const store = mockStore({
  userData: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songsData: {
    loading: false,
    songs: [],
    error: "",
  },
  playlistData: {
    loading: false,
    playlists: [],
    error: "",
  },
});

it("should render entire ErrorPage Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <ErrorPage />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
