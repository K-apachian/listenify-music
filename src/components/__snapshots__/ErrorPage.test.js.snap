// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render entire ErrorPage Component 1`] = `ShallowWrapper {}`;
